test/functional/provider/define_spec.lua

```diff
@@ -333,6 +333,39 @@ local function function_specs_for(fn, sync, first_arg_factory, init)
           runx(sync, handler, on_setup)
         end)
       end)
+
+      describe('with range', function()
```

Comment:
@bfredl I tried to follow the other tests in this file, like command tests that have `describes` like  'with nargs/range', 'with nargs/count/bang/register' and 'with nargs/count/bang/register/eval'. 
Would it be strange to merge just these new ones or do I need to restructure the other tests in the file?